packages/chain-adapters/src/evm/EvmBaseAdapter.ts

```diff
@@ -459,80 +481,98 @@ export abstract class EvmBaseAdapter<T extends EvmChainId> implements IChainAdap
     receiverAddress,
     hex,
   }: BroadcastTransactionInput): Promise<string> {
-    await Promise.all([
-      assertAddressNotSanctioned(senderAddress),
-      receiverAddress !== CONTRACT_INTERACTION && assertAddressNotSanctioned(receiverAddress),
-    ])
-    return this.providers.http.sendTx({ sendTxBody: { hex } })
+    try {
+      await Promise.all([
+        assertAddressNotSanctioned(senderAddress),
+        receiverAddress !== CONTRACT_INTERACTION && assertAddressNotSanctioned(receiverAddress),
+      ])
+
+      const txHash = await this.providers.http.sendTx({ sendTxBody: { hex } })
+
+      return txHash
+    } catch (err) {
+      if ((err as Error).name === 'ResponseError') {
+        const response = await ((err as any).response as Response).json()
+        const error = JSON.parse(response.message)
+
+        return ErrorHandler(JSON.stringify(response), {
+          translation: 'chainAdapters.errors.broadcastTransactionWithMessage',
+          options: { message: error.message },
+        })
+      }
+
+      return ErrorHandler(err, {
+        translation: 'chainAdapters.errors.broadcastTransaction',
+      })
+    }
```

Comment:
q: Seeing all those error handlers is making me considering an HOC catching all errors from all methods and then handling them with a mapping, am I going too far? So we could simplify all the adapters source codes

It might be a wrong idea, very hard to maintain and maybe we have too much edge cases to handle in each adapters

Reply:
The idea is most should be the standard `ErrorHandler` which will log the error and provide the translation, but the in order to expose the error reason for broadcast as returned from the node we need to parse the payloads accordingly. Each chain type has a different format, so just chose to keep the logic inline. I would be inclined to leave as is for that reason and considering chain adapters is quite stable and not touched too frequently. Let me know if that makes sense to you or if you still have hesitations.

Reply:
> The idea is most should be the standard `ErrorHandler` which will log the error and provide the translation, but the in order to expose the error reason for broadcast as returned from the node we need to parse the payloads accordingly. Each chain type has a different format, so just chose to keep the logic inline. I would be inclined to leave as is for that reason and considering chain adapters is quite stable and not touched too frequently. Let me know if that makes sense to you or if you still have hesitations.

Na it makes complete sense, I'm happy with both, even thought what I propose is reducing the code from a line number point of view, your solution is also super straighforward and easy to understand:

- What I propose is more complexe and might lead to weird bugs
- Your solution needs a human to think about the error handling while implementing an handler

Both have their pros and cons, happy with your one, just wanted to discuss it a little bit 🙏